learn2track/batch_schedulers.py

```diff
@@ -256,6 +256,47 @@ def _next_batch(self, batch_count):
         return self._prepare_batch(self._indices)
 
 
+class TractographyBatchSchedulerWithProportionalSamplingFromSubjects(TractographyBatchScheduler):
+    """ Batch scheduler for streamlines dataset, where streamlines are sampled proportionally from each subject
+    """
+
+    @property
+    def indices_per_subject(self):
+        if self._indices_per_subject is None:
+            subject_offsets = self.dataset.streamlines_per_sujet_offset + [len(self.dataset)]
+            self._indices_per_subject = [np.arange(subject_offsets[i], subject_offsets[i+1]) for i in range(self.dataset.subjects)]
+        return self._indices_per_subject
+
+    @property
+    def batch_size_per_subject(self):
+        if self._batch_size_per_subject is None:
+            subject_relative_sizes = np.array(self.dataset.nb_streamlines_per_sujet) / sum(self.dataset.nb_streamlines_per_sujet)
+            self._batch_size_per_subject = np.round(subject_relative_sizes * self.batch_size)
+        return self._batch_size_per_subject
```

Comment:
We should make sure that `assert np.sum(self._batch_size_per_subject) == self.batch_size`